apps/condo/domains/contact/access/ContactRole.js

```diff
@@ -0,0 +1,61 @@
+// @ts-nocheck
+/**
+ * Generated by `createschema contact.ContactRole 'name:Text'`
+ */
+
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+const {
+    queryOrganizationEmployeeFor,
+    checkOrganizationPermission,
+} = require('@condo/domains/organization/utils/accessSchema')
+const get = require('lodash/get')
+const { getById } = require('@core/keystone/schema')
+const { STAFF } = require('@condo/domains/user/constants/common')
+
+async function canReadContactRoles ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin || user.isSupport) return {}
+
+    if (user.type !== STAFF) return false
+
+    return {
+        OR: [
+            { organization_is_null: true },
+            { organization: queryOrganizationEmployeeFor(user.id) },
+        ],
+    }
+}
+
+async function canManageContactRoles ({ authentication: { item: user }, originalInput, operation, itemId }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+    if (user.isAdmin || user.isSupport) return true
+
+    if (user.type !== STAFF) return false
+
+    let organizationId
+    if (operation === 'create') {
+        organizationId = get(originalInput, ['organization', 'connect', 'id'])
+    } else if (operation === 'update') {
+        if (!itemId) return false
+        organizationId = get(originalInput, ['organization', 'connect', 'id'])
+
+        if (!organizationId) {
+            const contactRole = await getById('ContactRole', itemId)
+            organizationId = get(contactRole, 'organization')
+        }
+    }
+
+    return checkOrganizationPermission(user.id, organizationId, 'canManageContacts')
```

Comment:
i think we need to use `checkPermissionInUserOrganizationOrRelatedOrganization` to give manage access to employees in related `from` organization, how it is done in `access/Contact`

Reply:
> i think we need to use `checkPermissionInUserOrganizationOrRelatedOrganization` to give manage access to employees in related `from` organization, how it is done in `access/Contact`

done

> Are we not going to make a special role like `canManageContactRoles`?

I think that editing roles without editing contacts is not make sense.

Reply:
`canManageContactRoles` implemented